src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -32,20 +32,23 @@ public class SearchService {
     public SearchResult<SearchHit> search(@RestQuery String q) {
         var result = session.search(Guide.class)
                 .select(SearchHit.class)
-                .where((f, root) -> {
-                    root.add(f.matchAll()); // Default
-                    if (q != null && !q.isBlank()) {
-                        root.add(f.simpleQueryString()
-                                .field("keywords").boost(10.0f)
-                                .field("title").boost(7.0f)
-                                .field("summary").boost(5.0f)
-                                .field("fullContent")
-                                .field("keywords_autocomplete").boost(1.0f)
-                                .field("title_autocomplete").boost(0.7f)
-                                .field("summary_autocomplete").boost(0.5f)
-                                .field("fullContent_autocomplete").boost(0.1f)
-                                .matching(q));
+                .where(f -> {
+                    if (q == null || q.isBlank()) {
+                        return f.matchAll();
                     }
+
+                    return f.bool().must(f.simpleQueryString()
+                            .field("title").boost(10.0f)
```

Comment:
I thought topics/keywords were things we wanted to match in priority, that's why the title boost was lower.

If you want to change this, you should probably change it for `title_autocomplete` as well?

Reply:
Keep in mind that we had a very naive solution for search until now. So keywords was basically: "let's add some additional words that are not in the title or preamble", and nothing more.
We could change the definition but that's what it is atm.

As for topics, it might make sense to boost them more in the future but I would suggest to do the tweaking when all the guides are indexed and we have a proper idea of what we want (and also maybe once the PageRank thing is in).